pkg/server/apiHandlerv2.go

```diff
@@ -78,22 +79,38 @@ func (h *apiHandlerv2) recordClient(u *uuid.UUID, r *http.Request) {
 	h.manager.WriteRequest(*u, b)
 }
 
-func (h *apiHandlerv2) checkCertAndRecord(w http.ResponseWriter, r *http.Request) *uuid.UUID {
-	// only uses the device cert
-	cert := getClientCert(r)
-	u, err := h.manager.DeviceCheckCert(cert)
-	if err != nil {
-		log.Printf("error checking device cert: %v", err)
+//checkCertAndRecord processes http.Request and extracts AuthContainer
+//checks for certs hash in AuthContainer and go through saved certs to check auth state
+//returns ProtectedPayload and device uuid
+func (h *apiHandlerv2) checkCertAndRecord(w http.ResponseWriter, r *http.Request) ([]byte, *uuid.UUID) {
```

Comment:
If I get this correctly, we are changing this in a few ways:

1. No longer check the client cert, but rather check the authcontainer hash
2. No longer return the UUID of the device from the cert, but rather return the authcontainer content _and_ the UUID

We probably should change the func name to be a bit more descriptive. Something like `retrieveAndValidateAuthContainerAndRecord` (that is ugly, but gets the idea across).